cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogCleaner.java

```diff
@@ -71,4 +75,18 @@ public void run() {
     LOG.info("File cleanup completed, Successful file deletes - {}. Failed file deletes - {}. Log Cleanup took {} ms",
              deleteCount, failureCount, (completionTime - startTime));
   }
+
+  private void deleteParentFoldersIfEmpty(String logFilePath) throws IOException {
+    Location logFileLocation = Locations.getLocationFromAbsolutePath(locationFactory, logFilePath);
+    Location folderLocation = Locations.getParent(logFileLocation);
+    while (Locations.LOCATION_COMPARATOR.compare(logsDirectoryLocation, folderLocation) != 0
```

Comment:
Why need to compare with the logsDirectoryLocation??

Reply:
The log folders are deleted in bottom-up manner. All the empty descendants a `logs` folder are deleted.
While deleting the folders bottom-up, we want to stop at `logs` folder.